online/src/app/services/product.service.ts

```diff
@@ -24,23 +23,13 @@ export class ProductService {
     return this.http.delete<void>(`${url}/products/${id}`);
   }
 
-  addToCart(productID: number): void {
-    let productOrders = this.productOrders.find(productOrder => productOrder.productId === productID);
-    if (productOrders === undefined) {
-      this.productOrders.push({ productId: productID, quantity: 1 });
-    } else {
-      productOrders.quantity += 1;
+  checkout(productOrders: ProductOrder[]): Observable<string> {
+    if (productOrders.length > 0) {
+      const data = { customer: localStorage.getItem('username'), products: productOrders };
+      productOrders = [];
+      return this.http.post(`${url}/orders`, data, { responseType: 'text' });
     }
-  }
-
-  getCartOrders(): ProductOrder[] {
-    return this.productOrders;
-  }
-
-  checkout(): Observable<string> {
-    const data = { customer: localStorage.getItem('username'), products: this.productOrders };
-    this.productOrders = [];
-    return this.http.post(`${url}/orders`, data, { responseType: 'text' });
+    return new Observable<string>();
```

Comment:
why not just throw an error if the length is lower than 1? generating obserables like this is just so ugh...